modules/recommended-etcd-practices.adoc

```diff
@@ -6,59 +6,42 @@
-`etcd_mvcc_db_total_size_in_use_in_bytes` which indicates the actual
-database usage after a history compaction, and
-`etcd_debugging_mvcc_db_total_size_in_bytes` which shows the database size
-including free space waiting for defragmentation. Instructions on defragging
-etcd can be found in the `Defragmenting etcd data` section.
+For large and dense clusters, etcd can suffer from poor performance if the keyspace grows too large and exceeds the space quota. Periodically maintain and defragment etcd to free up space in the data store. Monitor Prometheus for etcd metrics and defragment it when required; otherwise, etcd can raise a cluster-wide alarm that puts the cluster into a maintenance mode that accepts only key reads and deletes. Monitor these key metrics:
 
-Etcd writes data to disk, so its performance strongly depends on disk performance. Etcd
-persists proposals on disk. Slow disks and disk activity from other processes might cause long
-fsync latencies, causing etcd to miss heartbeats, inability to commit new proposals to the disk
-on time, which can cause request timeouts and temporary leader loss. It is highly recommended to
-run etcd on machines backed by SSD/NVMe disks with low latency and high throughput.
+* `etcd_server_quota_backend_bytes`, which is the current quota limit
+* `etcd_mvcc_db_total_size_in_use_in_bytes`, which indicates the actual database usage after a history compaction
+* `etcd_debugging_mvcc_db_total_size_in_bytes`, which shows the database size, including free space waiting for defragmentation
 
-Some of the key metrics to monitor on a deployed {product-title} cluster
-are p99 of etcd disk write ahead log duration and the number of etcd leader changes.
-Use Prometheus to track these metrics. `etcd_disk_wal_fsync_duration_seconds_bucket`
-reports the etcd disk fsync duration, `etcd_server_leader_changes_seen_total` reports
-the leader changes. To rule out a slow disk and confirm that the disk is reasonably fast,
-99th percentile of the `etcd_disk_wal_fsync_duration_seconds_bucket` should be less than 10ms.
+For more information about defragmenting etcd, see the "Defragmenting etcd data" section.
 
-Fio, a I/O benchmarking tool can be used to validate the hardware for etcd before or after
-creating the {product-title} cluster. Run fio and analyze the results:
+Because etcd writes data to disk and persists proposals on disk, its performance depends on disk performance. Slow disks and disk activity from other processes can cause long fsync latencies, which can cause etcd to miss heartbeats, not commit new proposals to the disk on time, and ultimately experience request timeouts and temporary leader loss. Be sure to run etcd on machines that are backed by SSD/NVMe disks with low latency and high throughput. Consider single-level cell (SLC) solid-state drives (SSDs), which provide 1 bit per memory cell, are durable and reliable, and are ideal for write-intensive workloads.
```

Comment:
```suggestion
Because etcd writes data to disk and persists proposals on disk, its performance depends on disk performance. Slow disks and disk activity from other processes can cause long fsync latencies, which can cause etcd to miss heartbeats, not commit new proposals to the disk on time, and ultimately experience request timeouts and temporary leader loss. Ensure you run etcd on machines that are backed by SSD or NVMe disks with low latency and high throughput. Consider single-level cell (SLC) solid-state drives (SSDs), which provide 1 bit per memory cell, are durable and reliable, and are ideal for write-intensive workloads.
```

Reply:
Up to you!